### PR TITLE
Add `WP_ENV_TESTS_MYSQL_PORT` / `.wp-env.json` `.env.tests.mysqlPort` option etc

### LINKED ISSUE #50677: [Env Package]: Access wp-env db port via env variable
Currently, connecting directly to the wp-env database requires grabbing the port number manually. 
* [Accessing MYSQL DB](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/getting-started-with-code-contribution.md#accessing-the-mysql-database)
* [Issue Discussion](https://github.com/WordPress/gutenberg/issues/50542)

It would be really nice to access the port number via environment variables like, we have for sites (WP_ENV_PORT and WP_ENV_TESTS_PORT) 

[workaround soln. ](https://github.com/WordPress/gutenberg/issues/50542):
```
docker ps -f ancestor=mariadb -f name=tests-mysql  --format='{{.Ports}}' | sed -E 's/.*:(.*)->.*/\\1/'
```

### PR DESCRIPTION
**Pull request by BrianHenryIE**
Fix: https://github.com/WordPress/gutenberg/issues/50677

## What?

Currently, the tests container's database port is assigned randomly each time wp-env is started. This PR adds the ability to set the MySQL port number in the `.wp-env.json` file or via a `WP_ENV_TESTS_MYSQL_PORT` environmental variable.

## Why?

I'd like to be able to use the database in unit tests.

## How?

By adding a config option to keep the port number static.

## Testing Instructions

Set `testsMysqlPort` in `.wp-env.json`
```
{
	"testsMysqlPort": 52666
}
```

```
$ cat .wp-env.json | jq '.testsMySqlPort = 52666' | sponge .wp-env.json`
$ npx wp-env start
...
MySQL for automated testing is listening on port 52666
```

```
$ WP_ENV_TESTS_MYSQL_PORT=56678 npx wp-env start
...
MySQL for automated testing is listening on port 56678
```

## Screenshots or screencast 

Screenshot 2024-04-24 at 2 59 05 PM

Screenshot 2024-04-24 at 2 59 19 PM

## Note

I tried to also add it as a setting `env.tests.mysqlPort` but I got a parsing error:

`✖ Invalid /Users/brian.henry/Sites/wpcs-autoload/.wp-env.json: "tests.mysqlPort" is not a configuration option.`

### CODE REVIEW

File: packages/env/lib/config/parse-config.js [hunk 1]
```diff
@@ -85,6 +87,7 @@ const DEFAULT_ENVIRONMENT_CONFIG = {
 	themes: [],
 	port: 8888,
 	testsPort: 8889,
+	testsMysqlPort: null,
```
Comment (ObliviousHarmony) [inline packages/env/lib/config/parse-config.js]: I view the top-level `port` and `testsPort` properties are more of a legacy thing. It's more intuitive to set configuration like this _only_ at the environment level. How would you feel about dropping the root `testsMysqlPort` in favor of just having `development.mysqlPort` and `tests.mysqlPort`?

File: packages/env/lib/build-docker-compose-config.js [hunk 2]
```diff
@@ -167,6 +167,7 @@ module.exports = function buildDockerComposeConfig( config ) {
 	// Set the default ports based on the config values.
 	const developmentPorts = `\${WP_ENV_PORT:-${ config.env.development.port }}:80`;
 	const testsPorts = `\${WP_ENV_TESTS_PORT:-${ config.env.tests.port }}:80`;
+	const testsMysqlPorts = `\${WP_ENV_TESTS_MYSQL_PORT:-${ config.env.tests.mysqlPort ?? '' }}:3306`;
```
Comment (ObliviousHarmony) [inline packages/env/lib/build-docker-compose-config.js]: If we're going to support customizing the port like this let's go ahead and also support the development environment's MySQL port too.

**Review by ObliviousHarmony — CHANGES_REQUESTED** [review]
This seems like a reasonable addition to `wp-env`, thanks!

I've left some comments :smile:

**Comment by BrianHenryIE** [comment]
To test, checkout this PR and create a project with its `package.json` referencing it:

```json
{
  "name": "gutenberg-61057-test",
  "devDependencies": {
    "@wordpress/env": "file:/path/to/checked/out/gutenberg/packages/env"
  }
}
```

```bash
npm install;
npx wp-env start;
```

Observe default behaviour with randomly assigned MySQL ports:

```
MySQL is listening on port 54916
MySQL for automated testing is listening on port 54922
```

Add `.wp-env.json`:
```json
{
  "env": {
    "development": {
      "mysqlPort": 13306
    },
    "tests": {
      "mysqlPort": 13307
    }
  }
}
```

```bash
npx wp-env start;
```

Observe specified ports used:

```
MySQL is listening on port 13306
MySQL for automated testing is listening on port 13307
```

Test Bash env variables:

```bash
WP_ENV_MYSQL_PORT=13304 npx wp-env start
```

```
MySQL is listening on port 13304
MySQL for automated testing is listening on port 13307
```

```bash
WP_ENV_TESTS_MYSQL_PORT=13305 npx wp-env start
```

```
MySQL is listening on port 13306
MySQL for automated testing is listening on port 13305
```

~~I'll take another look at the automated tests soon. At first glance the failures don't seem related to these changes but are obviously wp-env config parsing failures.~~ Fixed.

**Review by desrosj — APPROVED** [review]
This seems sensible to me. Thanks for adding tests! @ObliviousHarmony has this been updated to your liking?

**Comment by desrosj** [comment]
Actually on second look I [mistook these changes as new tests](https://github.com/WordPress/gutenberg/pull/61057/files#diff-90b1c99fd7324cd750e05b3166ed8f0dc47fd4cec7925976a703e1065b9d364c). It would be great to add a few if you could.

**Comment by desrosj** [comment]
It seems like all of the feedback has been reasonably addressed. Since this is only being blocked by the review points to be confirmed as addressed, I'm going to merge this to give it a try.

**Pull request merged**